presets/ragengine/vector_store/base.py

```diff
@@ -151,12 +152,12 @@ def query(self,
             "metadata": query_result.metadata,
         }
 
-    def add_document_to_index(self, index_name: str, document: Document, doc_id: str):
+    async def add_document_to_index(self, index_name: str, document: Document, doc_id: str):
         """Common logic for adding a single document."""
         if index_name not in self.index_map:
             raise ValueError(f"No such index: '{index_name}' exists.")
         llama_doc = LlamaDocument(text=document.text, metadata=document.metadata, id_=doc_id)
-        self.index_map[index_name].insert(llama_doc)
+        await self.index_map[index_name].insert(llama_doc)
```

Comment:
This is in-memory operation, right? Do we need await?

Reply:
i think this one is not awaitable.

Reply:
this is llamaindex insert - llama_index/core/indices/base.py
yes it is not awaitable

```
    def insert(self, document: Document, **insert_kwargs: Any) -> None:
        """Insert a document."""
        with self._callback_manager.as_trace("insert"):
            nodes = run_transformations(
                [document],
                self._transformations,
                show_progress=self._show_progress,
                **insert_kwargs,
            )

            self.insert_nodes(nodes, **insert_kwargs)
            self.docstore.set_document_hash(document.get_doc_id(), document.hash)
```

Reply:
fixed